File: sentimental-cash/cash.py
from cs50 import get_float
changes_list = {'quarters' : 25,
                'dimes'    : 10,
                'nickles'  : 5,
                'pennies'  : 1}
# changes = [25,10,5,1]
while True:
    get_changes = get_float("change owed: ")
    get_changes = int(get_changes*100)
    if get_changes < 0:
        continue
    else:
        break
count_changes = 0
while get_changes != 0:
    if get_changes <= 0:
       break
    else:
        for i in changes_list.values():
            if i <= get_changes:
                if get_changes <= 0:
                    break
                get_changes = get_changes-i
                count_changes += 1
                break
print(count_changes)

File: cash/cash.c
// Inport necessary library
#include <stdio.h>
#include <cs50.h>

// Declaring a prototype for function
int count_coin(int change_owed);
int main(void)
{

// Get change owed from the customer
while (1){
    int change_owed = get_int("Change owed: ");
    if (change_owed < 0)
    {
        continue;
    }
    else
    {
        printf("%i\n",count_coin(change_owed));
        break;
    }

}
// Defining a function.
}
int count_coin(int change_owed)
{
    // declare required variable and set their value
    int quarters = 25;
    int dimes = 10;
    int nickles = 5;
    int pennies = 1;
    int count = 0;
    // Check if the change owed is among the variables if so try making the change_owed to zero and add one to count
    // if (change_owed == quarters || change_owed == dimes || change_owed == nickles || change_owed == pennies)
    // {
    //     count ++;
    //     return count;
    // }

    // Run while loop until change_owed is not equal to zero
    while (change_owed != 0)
    {
        // printf("%i\n",change_owed);
    if (change_owed <= 0)
    {
        break;
    }
        // if not try to minimize the change owed and add one to count
        if (change_owed >= quarters)
        {
            change_owed -= quarters;
            count++;
        }
        else if (change_owed >= dimes)
        {
            change_owed -= dimes;
            count++;
        }
        else if (change_owed >= nickles)
        {
            change_owed -= nickles;
            count++;
        }
        else if (change_owed >= pennies)
        {
            change_owed -= pennies;
            count++;
        }

    }
// When finally change owed is zero return count
    return count;
}

